x/foundation/foundation.go

```diff
@@ -0,0 +1,383 @@
+package foundation
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/line/lbm-sdk/codec"
+	codectypes "github.com/line/lbm-sdk/codec/types"
+	sdk "github.com/line/lbm-sdk/types"
+	sdkerrors "github.com/line/lbm-sdk/types/errors"
+)
+
+func DefaultDecisionPolicy(config Config) DecisionPolicy {
+	policy := &ThresholdDecisionPolicy{
+		Threshold: config.MinThreshold,
+		Windows: &DecisionPolicyWindows{
+			VotingPeriod: 24 * time.Hour,
+		},
+	}
+
+	// check whether the default policy is valid
+	if err := policy.ValidateBasic(); err != nil {
+		panic(err)
+	}
+	if err := policy.Validate(config); err != nil {
+		panic(err)
+	}
+
+	return policy
+}
+
+func validateProposers(proposers []string) error {
+	if len(proposers) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("no proposers")
+	}
+
+	addrs := map[string]bool{}
+	for _, proposer := range proposers {
+		if addrs[proposer] {
+			return sdkerrors.ErrInvalidRequest.Wrapf("duplicated proposer: %s", proposer)
+		}
+		addrs[proposer] = true
+
+		if err := sdk.ValidateAccAddress(proposer); err != nil {
+			return sdkerrors.ErrInvalidAddress.Wrapf("invalid proposer address: %s", proposer)
+		}
+	}
+
+	return nil
+}
```

Comment:
I cannot found the codes that verify the signatures of multi proposers.
Where is it?

Reply:
The module does not verify the signatures, but the antehandler does.